### DISCUSSION TITLE
firmware issue for sound, wifi and gpu on hp omen laptop

### DISCUSSION THREAD

**Discussion by Senshi111** [SpiralLinux/SpiralLinux-project#145] [General]
first thanks for spiral
ok. when i first install spiral , there was no sound, wifi, intel gpu detection
![2022-10-08_18-02_1](https://user-images.githubusercontent.com/40372242/194725441-a6105f3a-bd8d-411f-ab08-f2e5082f6f86.png)

![2022-10-08_18-02](https://user-images.githubusercontent.com/40372242/194725455-aca67075-0654-485a-9b80-79f6f8ae892c.png)

![2022-10-08_18-04](https://user-images.githubusercontent.com/40372242/194725465-a879165d-93d4-418b-9907-07cd6332d11c.png)

after i upgrade to debian sid, sound and mic are detected but still no intel gpu and wifi

![2022-10-08_18-18-1](https://user-images.githubusercontent.com/40372242/194725498-d9a0670d-de21-4ca5-aac9-51cc74bdabb3.png)

![2022-10-08_18-18-2](https://user-images.githubusercontent.com/40372242/194725508-c5784a7b-5dfa-4b79-b837-50c430a46907.png)

after i manually add i915 firmware my intel gpu is detected but still no wifi 

![2022-10-08_18-20-3](https://user-images.githubusercontent.com/40372242/194725572-c9c174fb-d940-475c-9d68-193775a3c5f6.png)

![2022-10-08_21-50-4](https://user-images.githubusercontent.com/40372242/194725581-bc497106-b2ba-4517-9186-706f6ef4c512.png)

after i add those iwlwifi firmware i got wifi working, dont know witch one fixed my wifi i didnt wanted to check so i past all that i have

![2022-10-08_21-55-5](https://user-images.githubusercontent.com/40372242/194725770-40d6b6d0-03d0-47bc-b623-a10cc810cae4.png)

![2022-10-08_21-56-6](https://user-images.githubusercontent.com/40372242/194725776-8f62025a-dda1-4d3f-9d71-aae7a37e2d7f.png)

just if someone have the same problem or if you can add those extra firmware

**Comment by CicadaSeventeen** [comment]
Even backports of debian stable may not be new enough for 12th core laptop. Could you tell me where do you get these firmware?

**Comment by CicadaSeventeen** [comment]
Debian Sid actually does not have most up-to-date firmware too. It's firmware is like several months newer than backports.
You can install Sid's firmware from Xanmod unofficial repository. Also there are newer kernel like 6.0. Newer kernel may also work for such problems. You can try to install it from a out-of-box SpiralLinux to check if it works.
see [here](https://www.linuxcapable.com/how-to-install-xanmod-linux-kernel-on-debian-11-bullseye/)
Notice that even after installing new kernel, the default kernel for Spiral is still 5.18. You need to go to `advance` menu in grub to boot from it.

**Reply by Senshi111** [reply]
yeah i already have xanmod kernel 6.0 it also fix my display refresh rate, but not all is fixed as far as wifi, intel gpu, sound and mic. i had to add those firmware manually

**Reply by CicadaSeventeen** [reply]
> yeah i already have xanmod kernel 6.0 it also fix my display refresh rate, but not all is fixed as far as wifi, intel gpu, sound and mic. i had to add those firmware manually

Have you tried to use firmware in xanmod repository?

**Reply by Senshi111** [reply]
Not now because added firmware myself. I can tell when i was playing around with some debian based distros, most of them have firmware problem for me because of 12th gen intel. Kubuntu 22.10 worked out of the box, sound, mic, Intel gpu , wifi.
So when i installed xanmod 6.0 on those distros that i was playing around, on some it fixed sound, mic, on some just sound but no mic but not fixed  wifi and Intel gpu.

**Reply by CicadaSeventeen** [reply]
> Not now because added firmware myself. I can tell when i was playing around with some debian based distros, most of them have firmware problem for me because of 12th gen intel. Kubuntu 22.10 worked out of the box, sound, mic, Intel gpu , wifi. So when i installed xanmod 6.0 on those distros that i was playing around, on some it fixed sound, mic, on some just sound but no mic but not fixed wifi and Intel gpu.

I see. So if you want to help you can count what firmware you have added. Maybe backports is not enough and we have to add new firmware package.

**Reply by Senshi111** [reply]
Sure i want to help, i will count those firmware and past it so you guys can see

**Reply by CicadaSeventeen** [reply]
> Sure i want to help, i will count those firmware and past it so you guys can see

Actually I am just a passer-by wanting to help. I think this project really cool but now there are too many problem here.

**Reply by Senshi111** [reply]
Yeah but it's good posting all those informations, it may help someone

**Reply by CicadaSeventeen** [reply]
It starts that I want to choose a stable distro and confused about whether to use debian or openSuSE leap. oS is not KISS, a lot of strange behaviors, but it can easy to use out-of-box and it have great btrfs support. For debian maybe I have to use `debootstrap` to install it if I want WIFI and BTRFS subvols ..

**Reply by Senshi111** [reply]
I was using before spiral, opensuse tumbleweed but i always like debian distros but nit too many rolling releases of debian sid. I tried siduction but had some problems with xanmod kernel also firmware. I tried regularly installing debian sid i have to manually config my btrfs during installation for timeshift, lot of pain in but. So i decided to go with spiral now.

**Reply by CicadaSeventeen** [reply]
So would you like to like ... to start a team? It seems that the only one developer here is the developer of GeckoLinux. I think SpiralLinux is actually much bigger program then GeckoLinux and there is less way to deal with such problems with one person. I also have friend who works on polishing debian but her goal is majorly providing good `wine` and `waydroid` experience. She is also working on her own as the only developer. I think a team work can be much better here.

And of course it need the creator of SpiralLinux project to make a decision finally. So please tell us your opinion please @geckolinux ?

**Reply by Senshi111** [reply]
If i can help anyhow sure

**Reply by CicadaSeventeen** [reply]
> If i can help anyhow sure

Nice. I find you know about firmware and Nvidia well. This is just waht we need. Then the only problem is the opinion of @geckolinux . I really like the idea of this project, but there are truly a lot to do, so I think we need a team that I can join in.

**Reply by geckolinux** [reply]
Hi there @CicadaSeventeen , all ideas for improvements and solutions are welcome, as long as they are beneficial and necessary configuration changes and/or package selections that can be cleanly integrated into SpiralLinux. No custom packages or modifications of Debian packages will be integrated into SpiralLinux since the main project goal is to remain completely dependent on Debian.

**Reply by CicadaSeventeen** [reply]
> Hi there @CicadaSeventeen , all ideas for improvements and solutions are welcome, as long as they are beneficial and necessary configuration changes and/or package selections that can be cleanly integrated into SpiralLinux. No custom packages or modifications of Debian packages will be integrated into SpiralLinux since the main project goal is to remain completely dependent on Debian.

OK thanks.

**Comment by Senshi111** [comment]
sure, i got them from here [https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/]([https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/](https://github.com/orgs/SpiralLinux/discussions/url))

**Comment by CicadaSeventeen** [comment]
BTW, if you do not mind you can rename this discussion to like 'firmware issue for sound, wifi and gpu' or something like that, so that we can guide those who need it into this discussion and we can keep discussing here.

**Comment by geckolinux** [comment]
Thanks for reporting this. I would usually recommend installing this DEB package from Testing or Unstable if the Intel WiFI firmware is too old:
https://packages.debian.org/bookworm/firmware-iwlwifi
Note that you might need to fully power off the system for a minute for it take effect.

If that still doesn't work then manually installing the firmware files as described here might be an option.
